activesupport/lib/active_support/cache/coder.rb

```diff
@@ -56,15 +56,10 @@ def load(dumped)
         version = load_version(dumped.byteslice(PACKED_VERSION_INDEX, version_length)) if version_length >= 0
         payload = dumped.byteslice((PACKED_VERSION_INDEX + [version_length, 0].max)..)
 
-        payload = @compressor.inflate(payload) if type & COMPRESSED_FLAG > 0
+        compressor = @compressor if type & COMPRESSED_FLAG > 0
+        serializer = STRING_DESERIALIZERS.fetch(type & ~COMPRESSED_FLAG) { @serializer }
```

Comment:
```suggestion
        serializer = STRING_DESERIALIZERS.fetch(type & ~COMPRESSED_FLAG, @serializer)
```

nitpick / micro-opt might as well save the block call.